bindings/python/scripts/generate.py

```diff
@@ -1,105 +1,318 @@
+                                .replace("struct ", "")
+                                .replace("pcl::", "")
+                            )
+                            argument_type_list.append(f"{type_ref} &")
+                elif sub_item["element_type"] == "Elaborated":
+                    namespace_ref = ""
+                    for sub_sub_item in sub_item["members"]:
+                        if sub_sub_item["kind"] == "NAMESPACE_REF":
+                            namespace_ref += f'{sub_sub_item["name"]}::'
+                        if sub_sub_item["kind"] == "TYPE_REF":
+                            argument_type_list.append(
+                                f'{namespace_ref}{sub_sub_item["name"]}'
+                            )
+                elif sub_item["element_type"] in ("Float", "Int"):
+                    argument_type_list.append(f'{sub_item["element_type"].lower()}')
+                else:
+                    argument_type_list.append(f'{sub_item["element_type"]}')
+        argument_type_list = ",".join(argument_type_list)
 
-    module_linelist.append(f'py::class_<{name_impl}>(m, "{name_decl}")')
-    for sub_item in item["members"]:
-        type_functions[sub_item["type"]](sub_item)
-    module_linelist.append(";")
-    in_struct = False
+        self.linelist.append(f".def(py::init<{argument_type_list}>())")
 
+    def handle_class_template(self):
+        flag = False
 
-def handle_macro(item):
-    pass
+        # TODO: Use list based method, like in handle_struct_decl
+        for sub_item in self.members:
+            if sub_item["kind"] == "TEMPLATE_NON_TYPE_PARAMETER":
+                self.linelist.append(
+                    f'template< {sub_item["element_type"].lower()} {sub_item["name"]} >'
+                )
+                flag = True
+        if not flag:
+            self.linelist.append(f"template<>")
+        cxx_base_specifier_list = [
+            sub_item["name"]
+            for sub_item in self.members
+            if sub_item["kind"] == "CXX_BASE_SPECIFIER"
+        ]
+        if cxx_base_specifier_list:
+            cxx_base_specifier_list = ",".join(cxx_base_specifier_list)
+            self.linelist.append(
+                f'py::class_<{self.name, cxx_base_specifier_list}>(m, "{self.name}")'
+            )
+        else:
+            self.linelist.append(f'py::class_<{self.name}>(m, "{self.name}")')
 
+    def handle_inclusion_directive(self):
+        # TODO: update blacklist
+        # blacklist = ["pcl/memory.h", "pcl/pcl_macros.h",]
+        if self.name.startswith("pcl"):
+            self.inclusion_list.append(self.name)
 
-type_functions = {
-    "include": handle_include,
-    "namespace": handle_namespace,
-    "alias": handle_alias,
-    "struct": handle_struct,
-    "constructor": handle_constructor,
-    "operator": handle_operator,
-    "macro": handle_macro,
-}
+    def handle_final(self, filename, module_name):
+        final = []
```

Comment:
No need to append when you can directly put them in the array

Reply:
Taken care of.